### ISSUE TITLE
deleting constructors/assignment operators should not invoke warning about not changing the other constructors.

### ISSUE THREAD

**Issue by sbromberger** [llvm/llvm-project#92954]
Consider the following code:
```
struct S {                                                                                                            
    S(const S &) = delete;
    void operator=(const S &) = delete;
};
```

`clang-tidy` (version 17.0.6) warns:
> Class 'S' defines a copy constructor and a copy assignment operator but does not define a destructor, a move constructor or a move assignment operator (cppcoreguidelines-special-member-functions)

This seems to me to be incorrect, in that only _deleting_ constructors/assignment operators should not result in a violation of the Rule of 5.

**Comment by EugeneZelenko** [comment]
Could you please try 18 or `main` branch?

**Comment by sbromberger** [comment]
> Could you please try 18 or `main` branch?

Sorry - I can't install other versions on the system I'm using. Is there some other way I can help? Thanks.

**Comment by PiotrZSL** [comment]
`AllowMissingMoveFunctionsWhenCopyIsDeleted`  option helpfs for missing move, but check will still complain about destructor.

**Comment by MitalAshok** [comment]
Here it is with clang-tidy trunk on Compiler Explorer: https://godbolt.org/z/f9fo53qYx

The core guideline this is based on does seem to suggest you have to write `~S() = default;`: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-five

The hicpp check alias also seems to require it too: https://www.perforce.com/resources/qac/high-integrity-cpp-coding-standard/special-member-functions#paragraph-28726

https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/special-member-functions.html
https://clang.llvm.org/extra/clang-tidy/checks/hicpp/special-member-functions.html

**Comment by dwblaikie** [comment]
Yeah, my 2c would be this isn't a bug and the warning is suitable. The reason for the rule of 0/3/5 is to reduce the chance of bugs - one reason folks turn off the copy ctor/assignment is because, for instance, their type holds some resource that must be cleaned up (and copying handles to the resource could lead to duplicate cleanup) - so the dtor should be considered along with the copy operations (even when only removing the copy operations).
